File: models/userModel.js
const pool = require('../config/db');

const createUser = (reg_no, email, phone, full_name, password_hash, campus, role = 'student') => {
    return pool.query(
        'INSERT INTO users (reg_no, email, phone, full_name, password_hash, role, campus) VALUES (?, ?, ?, ?, ?, ?, ?)',
        [reg_no, email, phone || null, full_name, password_hash, role, campus]
    );
};

const findByEmail = (email) => {
    return pool.query('SELECT * FROM users WHERE email = ?', [email]);
};

const findById = (id) => {
    return pool.query('SELECT id, reg_no, email, phone, full_name, role, campus, course FROM users WHERE id = ?', [id]);
};

const findByPhone = (phone) => {
    return pool.query('SELECT * FROM users WHERE phone = ?', [phone]);
};

const updatePassword = (id, password_hash) => {
    return pool.query('UPDATE users SET password_hash = ? WHERE id = ?', [password_hash, id]);
};

const updateProfile = (id, full_name, course, phone, email) => {
    if (email) {
        return pool.query('UPDATE users SET full_name = ?, course = ?, phone = ?, email = ? WHERE id = ?', [full_name, course || null, phone || null, email, id]);
    }
    return pool.query('UPDATE users SET full_name = ?, course = ?, phone = ? WHERE id = ?', [full_name, course || null, phone || null, id]);
};

const findByRole = (...roles) => {
    return pool.query(`SELECT id FROM users WHERE role IN (${roles.map(() => '?').join(',')})`, roles);
};

const findNameById = (id) => {
    return pool.query('SELECT full_name, reg_no FROM users WHERE id = ?', [id]);
};

module.exports = { createUser, findByEmail, findById, findByPhone, updatePassword, updateProfile, findByRole, findNameById };

File: zetech_hostel_app/lib/models/user_model.dart
class User {
  final String? id;
  final String? regNo;
  final String? email;
  final String? fullName;
  final String? role;
  final String? campus;
  final String? phone;
  final String? course;

  User({this.id, this.regNo, this.email, this.fullName, this.role, this.campus, this.phone, this.course});

  factory User.fromJson(Map<String, dynamic> json) => User(
    id: (json['_id'] ?? json['id'])?.toString(),
    regNo: (json['regNo'] ?? json['reg_no'])?.toString(),
    email: json['email']?.toString(),
    fullName: (json['fullName'] ?? json['full_name'])?.toString(),
    role: json['role']?.toString(),
    campus: json['campus']?.toString(),
    phone: json['phone']?.toString(),
    course: json['course']?.toString(),
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'regNo': regNo,
    'email': email,
    'fullName': fullName,
    'role': role,
    'campus': campus,
    'phone': phone,
    'course': course,
  };
}
